src/PassportServiceProvider.php

```diff
@@ -34,13 +35,61 @@ class PassportServiceProvider extends ServiceProvider
      */
     public function boot()
     {
-        $this->loadViewsFrom(__DIR__.'/../resources/views', 'passport');
+        $this->registerRoutes();
+        $this->registerResources();
+        $this->registerMigrations();
+        $this->registerPublishing();
+        $this->registerCommands();
 
         $this->deleteCookieOnLogout();
+    }
 
-        if ($this->app->runningInConsole()) {
-            $this->registerMigrations();
+    /**
+     * Register the Passport routes.
+     *
+     * @return void
+     */
+    protected function registerRoutes()
+    {
+        Route::group([
```

Comment:
@driesvints hello, why did you not move the group declaration to `web.php`?

In that case you can avoid useless `Route::group` declaration if routes are cached (condition inside `$this->loadRoutesFrom()`).

And this method will look like:
```
protected function registerRoutes()
{
    $this->loadRoutesFrom(__DIR__.'/../routes/web.php');
}
```
or we can remove this method wrapper and call directly `$this->loadRoutesFrom(__DIR__.'/../routes/web.php');`

What do you think?

Reply:
We don't do this mostly in our packages. It's fine I think.

Reply:
Yes, I've checked all laravel packages with route and think that the core team avoid using `config` inside `routes` files, but I saw that you it. And also we have duplicated conditions in that way.

For example horizon package:
https://github.com/laravel/horizon/blob/5.x/src/HorizonServiceProvider.php#L54

we have the condition which already exists inside `loadRoutesFrom` method

Reply:
@siarheipashkevich this condition was added in https://github.com/laravel/horizon/pull/1367 only recently.

Reply:
@driesvints thanks, but I think we should optimize this for avoiding extra time of code execution.

Reply:
It's fine really.